docs/resources/task.md

```diff
@@ -0,0 +1,224 @@
+3. Run the given `script` upon completion or `timeout` in the cloud
+
+## Example Usage
+
+```hcl
+resource "iterative_task" "task" {
+  name  = "example"
+  cloud = "aws"
+
+  environment = {GREETING = "Hello, world!"}
+  directory   = "${path.root}/shared"
+
+  script = <<-END
+    #!/bin/bash
+    echo "$GREETING" | tee $(uuidgen)
+  END
+}
+```
+
+## Argument Reference
+
+### Required
+
+- `name` - (Required) Task name.
+- `cloud` - (Required) Cloud provider to run the task on; valid values are `aws`, `gcp`, `az` and `k8s`.
+- `script` - (Required) Script to run; must begin with a valid [shebang](<https://en.wikipedia.org/wiki/Shebang_(Unix)>).
+
+### Optional
+
+- `region` - (Optional) [Cloud region/zone](#cloud-regions) to run the task on.
+- `machine` - (Optional) See [Machine Types](#machine-types) below.
+- `disk_size` - (Optional) Size of the ephemeral machine storage.
+- `spot` - (Optional) Spot instance price. `-1`: disabled, `0`: automatic price, any other positive number: fixed price.
+- `image` - (Optional) [Machine image](#machine-images) to run the task with.
+- `parallelism` - (Optional) Number of machines to be launched in parallel.
+- `directory` - (Optional) Local directory to synchronize.
+- `environment` - (Optional) Map of environment variable names and values for the task script. Empty string values are replaced with local environment values.
+- `timeout` - (Optional) Maximum number of seconds to run before termination.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+- `ssh_public_key` - Used to access the created machines.
+- `ssh_private_key` - Used to access the created machines.
+- `addresses` - IP addresses of the currently active machines.
+- `status` - Status of the machine orchestrator.
+- `events` - List of events for the machine orchestrator.
+- `logs` - List with task logs; one for each machine.
+
+~> **Note**: Status and events don't produce a stable output between cloud providers and are intended for human consumption only.
+
+## Machine Type
+
+### Generic
+
+The Iterative Provider offers some common machine sizes which are roughly the same
+for all supported clouds.
+
+#### No GPU
+
+- `m` - Medium, with (at least), 8 CPU cores and 16 GB of RAM.
+- `l` - Large, with (at least) 32 CPU cores and 64 GB of RAM.
+- `xl` - Extra large, with (at least) 64 CPU cores and 128 GB of RAM.
+
+#### NVIDIA Tesla K80 GPU
+
+- `m+k80` - Medium, with (at least) 4 CPU cores, 53 GB of RAM and 1 GPU device.
+- `l+k80` - Large, with (at least) 12 CPU cores, 112 GB of RAM and 2 GPU devices.
+- `xl+k80` - Extra large, with (at least) 24 CPU cores, 212 GB of RAM and 4 GPU devices.
+
+#### NVIDIA Tesla V100 GPU
+
+- `m+v100` - Medium, with (at least) 4 CPU cores, 61 GB of RAM and 1 GPU device.
+- `l+v100` - Large, with (at least) 12 CPU cores, 224 GB of RAM and 2 GPU devices.
+- `xl+v100` - Extra large, with (at least) 24 CPU cores, 448 GB of RAM and 4 GPU devices.
```

Comment:
> _based on `iterative/*/provider.go`_

Did you mean `task/**.go`? 😉

Reply:
I'm inclined to think that people who care about the exact machine type will also specify whatever type they want instead of using those common “sizes” though.

Reply:
| Type      | Minimum                                                | `aws`         | `az`                | `gcp`                                             | `k8s`                                                  |
| --------- | ------------------------------------------------------ | ------------- | ------------------- | ------------------------------------------------- | ------------------------------------------------------ |
| `m`       | 8 CPU cores<br>16 GB of RAM                            | `m5.2xlarge`  | `Standard_F8s_v2`   | `e2-custom-8-32768`                               | `cpu: 8`<br>`memory: 32G`                              | 
| `l`       | 32 CPU cores<br>64 GB of RAM                           | `m5.8xlarge`  | `Standard_F32s_v2`  | `e2-custom-32-131072`                             | `cpu: 32`<br>`memory: 128G`                            | 
| `xl`      | 64 CPU cores<br>128 GB of RAM                          | `m5.16xlarge` | `Standard_F64s_v2`  | `n2-custom-64-262144`                             | `cpu: 64`<br>`memory: 256G`                            | 
| `m+k80`   | 4 CPU cores<br>53 GB of RAM<br>1 × NVIDIA Tesla K80    | `p2.xlarge`   | `Standard_NC6`      | `custom-8-53248`<br>1 × `nvidia-tesla-k80`        | `cpu: 4`<br>`memory: 64G`<br>1 × `nvidia-tesla-k80`    | 
| `l+k80`   | 12 CPU cores<br>112 GB of RAM<br>2 × NVIDIA Tesla K80  | `p2.8xlarge`  | `Standard_NC12`     | `custom-32-131072`<br>4 × `nvidia-tesla-k80`      | `cpu: 32`<br>`memory: 512G`<br>8 × `nvidia-tesla-k80`  | 
| `xl+k80`  | 24 CPU cores<br>212 GB of RAM<br>4 × NVIDIA Tesla K80  | `p2.16xlarge` | `Standard_NC24`     | `custom-64-212992-ext`<br>8 × `nvidia-tesla-k80`  | `cpu: 64`<br>`memory: 768G`<br>16 × `nvidia-tesla-k80` | 
| `m+v100`  | 4 CPU cores<br>61 GB of RAM<br>1 × NVIDIA Tesla V100   | `p3.xlarge`   | `Standard_NC6s_v3`  | `custom-8-65536-ext`<br>1 × `nvidia-tesla-v100`   | `cpu: 8`<br>`memory: 64G`<br>1 × `nvidia-tesla-v100`   | 
| `l+v100`  | 12 CPU cores<br>224 GB of RAM<br>2 × NVIDIA Tesla V100 | `p3.8xlarge`  | `Standard_NC12s_v3` | `custom-32-262144-ext`<br>4 × `nvidia-tesla-v100` | `cpu: 32`<br>`memory: 256G`<br>4 × `nvidia-tesla-v100` | 
| `xl+v100` | 24 CPU cores<br>448 GB of RAM<br>4 × NVIDIA Tesla V100 | `p3.16xlarge` | `Standard_NC24s_v3` | `custom-64-524288-ext`<br>8 × `nvidia-tesla-v100` | `cpu: 64`<br>`memory: 512G`<br>8 × `nvidia-tesla-v100` |

Reply:
Commented with the table above to disambiguate taste from laziness. 😉 I still believe that it's too much information for users who rely on default values, and pointless for those who don't.

Reply:
IMO the helper types should not contain a flood of information and just give general specs. I would include a footnote that links to the above table for reference but I don't think it should be present in the first explanation of `Type` options

Reply:
<kbd>won't “fix”</kbd> on this pull request, but added to https://github.com/iterative/terraform-provider-iterative/pull/237#issue-776178285 as a potential follow–up.

Reply:
sure - but that table is awesome, thx.